inc/saibuffer.h

```diff
@@ -484,21 +484,29 @@ typedef enum _sai_buffer_profile_attr_t
      * @brief Dynamic threshold for the shared usage
      *
      * The threshold is set to the 2^n of available buffer of the pool.
-     * Mandatory when SAI_BUFFER_POOL_THRESHOLD_MODE = SAI_BUFFER_THRESHOLD_MODE_DYNAMIC
```

Comment:
Kamil,
As I think we don't have the ability to check another object
SAI_BUFFER_PROFILE_ATTR_POOL_ID.SAI_BUFFER_POOL_ATTR_THRESHOLD_MODE
I think the current pull request is the best that can be done now
Can you approve ?